vassal-app/src/main/java/VASSAL/counters/BoundsTracker.java

```diff
@@ -29,25 +31,27 @@
  * added pieces belonged.
  */
 public class BoundsTracker {
-  private final Set<Map> maps;
+  private final List<Pair<Rectangle, Map>> regions;
 
   public BoundsTracker() {
-    maps = new HashSet<>();
+    regions = new LinkedList<>();
   }
 
   public void clear() {
-    maps.clear();
+    regions.clear();
   }
 
   public void addPiece(GamePiece p) {
     if (p.getMap() != null) {
-      maps.add(p.getMap());
+      final Rectangle region = p.boundingBox();
+      region.translate(p.getPosition().x, p.getPosition().y);
+      regions.add(new Pair<>(region, p.getMap()));
     }
   }
 
   public void repaint() {
-    for (final Map m : maps) {
-      m.repaint();
+    for (final Pair<Rectangle, Map> rmPair : regions) {
+      rmPair.second.repaint(rmPair.first);
```

Comment:
It reads a bit oddly to me to have the pair go from small to large; mabye put the `Map` first?

Reply:
Fair enough. In my head I was calling the tuple a "region-map" rather than a "map-region", so wanted the pair to be region then map. But the variable name ended up clearer as `rmPair` rather than `regionMap`, and there is no real difference between `rmPair` and `mrPair`.

Reply:
Well, other than future readers might think mrPair is "Mr. Pair" ;)

Reply:
I was working with a Map-Reduce application some years ago where _everything_ was Mr _Something_.